src/ngraph/op/non_zero.cpp

```diff
@@ -70,3 +74,98 @@ shared_ptr<Node> op::v3::NonZero::clone_with_new_inputs(const OutputVector& new_
     check_new_args_count(this, new_args);
     return make_shared<v3::NonZero>(new_args.at(0), m_output_type);
 }
+
+namespace
+{
+    template <element::Type_t INPUT_ET, element::Type_t OUT_ET>
+    bool evaluate_nonzero_execute(const HostTensorPtr& input, const HostTensorPtr& output)
+    {
+        using T = typename element_type_traits<INPUT_ET>::value_type;
```

Comment:
How about IN_T and OUT_T instead of T and U?

Reply:
sure